ghost/core/core/server/api/endpoints/previews.js

```diff
@@ -2,11 +2,39 @@ const tpl = require('@tryghost/tpl');
 const errors = require('@tryghost/errors');
 const models = require('../../models');
 const ALLOWED_INCLUDES = ['authors', 'tags'];
+const ALLOWED_MEMBER_STATUSES = ['anonymous', 'free', 'paid'];
 
 const messages = {
     postNotFound: 'Post not found.'
 };
 
+// Simulate serving content as different member states by setting the minimal
+// member context needed for content gating to function
+const _addMemberContextToFrame = (frame) => {
+    if (!frame?.options?.memberStatus) {
+        return;
+    }
+
```

Comment:
Would it make things safer if we add member validation here, eg checking that `frame.options.memberStatus` has one of `ALLOWED_MEMBER_STATUSES` and perhaps throw an error or default? In case

Reply:
Yeah, we could potentially add some extra safety here. Although in theory it should never be possible as the [API filters the memberStatus option](https://github.com/TryGhost/Ghost/pull/22168/files#diff-f5713173a4138c9fb6f492e66ff27cad61e8250ecce7f13e7b7e33f39cec3411R59-R60) before this code is reached.